src/lead_registration/dto/create-lead_registration.dto.ts

```diff
@@ -0,0 +1,22 @@
+import {IsString, IsIn, IsNotEmpty } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class CreateLeadRegistrationDto{
+  @ApiProperty({
+    description: 'Position the user is applying for',
+    example: 'Backend lead'
+  })
+  @IsString()
+  @IsNotEmpty()
+  role: string;
+
+  @ApiProperty({
+    description: 'status of registration', 
+    example: 'pending',
+    enum: ['pending', 'approved', 'rejected']
+  })
+  @IsString()
+  @IsNotEmpty()
+  @IsIn(['pending', 'approved', 'rejected'])
```

Comment:
@snebo we need to fine-tune, the linking of the leads to users, or will the leads be created to their own entity.

Reply:
@snebo what questions are we referring to here?

Reply:
The user role is already doing that

Reply:
And le'ts follow the same naming convention fro this file, update to "create-lead-registration.dto.ts"